pkg/autoscaler/stats_reporter.go

```diff
@@ -249,24 +246,19 @@ func (r *Reporter) ReportRequestedPodCount(v int64) error {
 	return r.report(requestedPodCountM.M(v))
 }
 
-// ReportActualPodCount captures value v for actual pod count measure.
-func (r *Reporter) ReportActualPodCount(v int64) error {
-	return r.report(actualPodCountM.M(v))
-}
-
-// ReportNotReadyPodCount captures value v for not ready pod count measure.
-func (r *Reporter) ReportNotReadyPodCount(v int64) error {
-	return r.report(notReadyPodCountM.M(v))
-}
-
-// ReportPendingPodCount captures value v for pending pod count measure.
-func (r *Reporter) ReportPendingPodCount(v int64) error {
-	return r.report(pendingPodCountM.M(v))
-}
-
-// ReportTerminatingPodCount captures value v for terminating pod count measure.
-func (r *Reporter) ReportTerminatingPodCount(v int64) error {
-	return r.report(terminatingPodCountM.M(v))
+// ReportActualPodCount captures values for ready, not ready, terminating, and pending pod count measure.
+func (r *Reporter) ReportActualPodCount(ready, notReady, terminating, pending int64) error {
+	var err error
+	if err = r.report(actualPodCountM.M(ready)); err != nil {
+		return err
+	}
```

Comment:
The only error in those is 
```
if !r.initialized {
		return errors.New("StatsReporter is not initialized yet")
	}
```
which means we can just check it at the beginning and then just call `report`.

Reply:
Or ditch this check altogether 😂

Reply:
I am all for ditching checks 😅

Reply:
This seems like an orthogonal cleanup and probably was added for a reason (not sure which, but 🤷‍♂ )

Reply:
you mean the PR or the check?

I asked @taragu to do the PR as a followup to this conversation: https://github.com/knative/serving/pull/6476#discussion_r365996047

Reply:
the check

Reply:
Should we ditch `r.initialized` altogether? It seems it's not being used elsewhere except for the check 
```
if !r.initialized {
		return errors.New("StatsReporter is not initialized yet")
	}
```

Reply:
Yeah afaik it doesn't do something useful.